pkg/spoke/submarineragent/config_controller.go

```diff
@@ -553,6 +581,80 @@ func (c *submarinerConfigController) updateGatewayStatus(ctx context.Context, re
 	return c.updateSubmarinerConfigStatus(ctx, recorder, config, &condition)
 }
 
+func (c *submarinerConfigController) setNetworkTypeIfAbsent(ctx context.Context, config *configv1alpha1.SubmarinerConfig,
+	recorder events.Recorder,
+) error {
+	if config.Status.ManagedClusterInfo.NetworkType != "" {
+		return nil
+	}
+
+	networks, err := c.dynamicClient.Resource(networksGVR).Get(context.TODO(), networksConfigName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	networkType, _, err := unstructured.NestedString(networks.Object, "spec", "networkType")
+	if err != nil {
+		return err
+	}
+
+	config.Status.ManagedClusterInfo.NetworkType = networkType
+	_, updated, updatedErr := submarinerconfig.UpdateStatus(ctx, c.configClient.SubmarineraddonV1alpha1().SubmarinerConfigs(config.Namespace),
+		config.Name, submarinerconfig.UpdateStatusFn(nil, &config.Status.ManagedClusterInfo))
+
+	if updated {
+		recorder.Eventf("SubmarinerConfigNetworkTypeSet",
+			"submarinerconfig network type was set to %s for managed cluster %s", networkType, config.Namespace)
+	}
+
+	return updatedErr
+}
+
+func (c *submarinerConfigController) validateOCPVersion(ctx context.Context, config *configv1alpha1.SubmarinerConfig,
+	recorder events.Recorder,
+) (bool, error) {
+	if config.Status.ManagedClusterInfo.Vendor != constants.ProductOCP {
+		// we only check OCP version if vendor is OCP
+		return true, nil
+	}
+
+	// We need OCP 4.11+ if using OVNK
+	err := c.setNetworkTypeIfAbsent(ctx, config, recorder)
+	if apiErrors.IsNotFound(err) {
+		// networks not found, deletion in progress, ignore
+		return false, nil
+	}
+
+	if err != nil {
+		return false, err
+	}
+
+	networkType := config.Status.ManagedClusterInfo.NetworkType
+	ocpVersion := config.Status.ManagedClusterInfo.VendorVersion
+
+	if networkType == cni.OVNKubernetes && version.Compare(version.Normalize(ocpVersion), "4.11.0", "<") {
+		ovnCondition := metav1.Condition{
+			Type:    "SubmarinerGatewayNodesLabeled",
```

Comment:
Can we use the existing constant - "submarinerGatewayCondition"

Reply:
Is it because you are updating the ManagedClusterAddon and not SubmarinerConfig?

Reply:
yeah - the `SubmarinerGatewayNodesLabeled` condition type checked in the UI. Ideally we shouldn't overload this condition. The `gatewaysStatusController` is responsible for maintaining this condition. It may be problematic also setting it here, ie if gateway nodes are still labeled then it could overwrite this setting. I think it would be safest (and perhaps cleaner) to put this code in the `gatewaysStatusController`.

Reply:
Upon further discussion, using the `SubmarinerGatewayNodesLabeled` condition is the only direct way to inform the user so, unfortunately, we're limited. Also we want to elide cloud prepare so we can't do the check in the  `gatewaysStatusController`.

Reply:
Let's use the constant:

```suggestion
			Type:    submarinerGatewayNodesLabeled,
```